src/v1/internal/connector.js

```diff
@@ -347,50 +341,37 @@ class Connection {
           }
         },
         onCompleted: () => {
-          this._ackFailureMuted = false;
           resolve();
         }
-      };
-      const queued = this._queueObserver(observer);
-      if (queued) {
-        this._packer.packStruct(RESET, [], err => this._handleFatalError(err));
-        this._chunker.messageBoundary();
-        this.sync();
-      }
+      });
     });
   }
 
-  _ackFailureIfNeeded() {
-    if (this._ackFailureMuted) {
-      return;
-    }
-
-    if (this._log.isDebugEnabled()) {
-      this._log.debug(`${this} C: ACK_FAILURE`);
-    }
-
-    const observer = {
+  _resetOnFailure() {
```

Comment:
Wdyt about calling `resetAndFlush` instead, like `resetAndFlush().finally(() => { /* clear failure */ })`?

Reply:
It would definitely make the code a bit more concise but then failure will be cleared in a tick after the one that processes the RESET response. **I think** it might result in some queued callbacks to be failed unnecessarily.